### PR TITLE
Add temp correction convsimfit

### LINKED ISSUE #26881: Convolution fit and simultaneous fitting
simultaneous fitting in the indirect data analysis convolution fit tab

### PR DESCRIPTION
**Pull request by Matthew-Andrew**
**Description of work.**

This adds a temperature correction to the convolution fitting tab. 

**To test:**
  * I have sent a package to relevant scientists for them to functionally test. 
  * Here code review is needed.
  * It would also be good to test edge case inputs to check that such cases are handled gracefully.
  * Relevant data can be found in #26631

Fixes #26881  

*This does not require release notes* because this is only going into a feature branch. A full set of release notes covering the simultaneous fitting will be done later.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by AnthonyLim23** [comment]
@Mantid-Matthew is this ready for @Harrietbrown to review?

**Comment by Mantid-Matthew** [comment]
This will be ready for re-review once #27880 is merged into master and this rebased.

**Comment by Mantid-Matthew** [comment]
@Harrietbrown Now that #27880 has been merged this should be ready to review Thanks Mat

**Review by Harrietbrown — CHANGES_REQUESTED** [review]

**Comment by Harrietbrown** [comment]
While not related to this issue spesificaly if the interface window is smaller (the default size on my pc) there will be a warning every time the fiting replots, this is done several times at the end of the fit. I don't know if this is worth it's own issue.

**Comment by Mantid-Matthew** [comment]
I have noticed this as well and on windows the plotting in general looks ill-sized. I don't know whether this is a windowing issue or a matplotlib version thing but agree that it should have an issue opened for it.

File: qt/scientific_interfaces/Indirect/IndirectFunctionBrowser/ConvFunctionModel.cpp [hunk 1]
```diff
@@ -77,56 +80,62 @@ void ConvFunctionModel::checkConvolution(IFunction_sptr fun) {
...
+  } else if (name == "DeltaFunction") {
+    m_hasDeltaFunction = true;
+  } else {
+    clear();
+    throw std::runtime_error(
+        "Function has wrong structure. Function not recognized");
   }
 }
 
+// void ConvFunctionModel::checkProductFunction(IFunction_sptr fun) {}
```
[the remark this answers is not in the document]
Comment (Mantid-Matthew) [reply qt/scientific_interfaces/Indirect/IndirectFunctionBrowser/ConvFunctionModel.cpp]: Thanks that shouldn't be there have removed.

**Review by Harrietbrown — APPROVED** [review]
I have created an issue for the plotting errors found with a small gui #27914.
Beyond that the code looks good and runs without other errors. Good Job.

**Comment by martyngigg** [comment]
Test failures are unrelated and fixed on master.

**Pull request merged**